invokeai/app/services/item_storage/item_storage_memory.py

```diff
@@ -0,0 +1,42 @@
+from typing import Generic, Optional, TypeVar
+
+from pydantic import BaseModel
+
+from invokeai.app.services.item_storage.item_storage_base import ItemStorageABC
+from invokeai.app.services.shared.pagination import PaginatedResults
+
+T = TypeVar("T", bound=BaseModel)
+
+
+class ItemStorageMemory(ItemStorageABC, Generic[T]):
```

Comment:
Any reason not to go ahead and swap this out in the dependencies?

Reply:
I suppose not - we'll hold off on merging this til we're ready anyways